CC/Sounder/receiver.cc

```diff
@@ -243,42 +246,40 @@ void Receiver::loopRecv(int tid, int core_id, SampleBuffer* rx_buffer)
-
-                    if (sf == 0)
-                        frame_id++;
-
-                    // only write received pilot into samp
-                    // otherwise use syncrxbuffBs as a dummy buffer
-                    if (config_->isPilot(frame_id, sf)) {
-                        r = baseRadioSet_->radioRx(it, samp, rxTimeBs);
-                        symbol_id = sf;
-                    } else {
-                        r = baseRadioSet_->radioRx(it, syncrxbuffBs.data(), rxTimeBs);
-                    }
+                if (symbol_id == config_->symbolsPerFrame) {
+                    symbol_id = 0;
+                    frame_id++;
+                }
 
-                    if (r < 0) {
-                        config_->running = false;
-                        break;
-                    }
-                    if (r != rx_len) {
-                        std::cerr << "BAD Receive(" << r << "/" << rx_len
-                                  << ") at Time " << rxTimeBs
-                                  << ", frame count " << frame_id << std::endl;
-                    }
+                int rx_len = config_->sampsPerSymbol;
+                int r;
 
-                    // schedule next beacon in 10 frames
-                    // TODO: make 10 a system-level param
-                    if (sf == 0) {
-                        txTimeBs = rxTimeBs + config_->sampsPerSymbol * config_->symbolsPerFrame * 10;
-                        int r_tx = baseRadioSet_->radioTx(it, beaconbuff.data(), 2, txTimeBs);
-                        if (r_tx != config_->sampsPerSymbol)
-                            std::cerr << "BAD Transmit(" << r_tx << "/" << config_->sampsPerSymbol
-                                      << ") at Time " << txTimeBs
-                                      << ", frame count " << frame_id << std::endl;
-                    }
+                // only write received pilot into samp
+                // otherwise use syncrxbuffBs as a dummy buffer
+                if (config_->isPilot(frame_id, symbol_id) || config_->isData(frame_id, symbol_id))
+                    r = baseRadioSet_->radioRx(it, samp, rxTimeBs);
+                else
+                    r = baseRadioSet_->radioRx(it, syncrxbuffBs.data(), rxTimeBs);
+
+                if (r < 0) {
+                    config_->running = false;
+                    break;
+                }
+                if (r != rx_len) {
+                    std::cerr << "BAD Receive(" << r << "/" << rx_len
+                              << ") at Time " << rxTimeBs
+                              << ", frame count " << frame_id << std::endl;
+                }
+
+                // schedule next beacon in 10 frames
+                // TODO: make 10 a system-level param
+                if (symbol_id == 0) {
+                    txTimeBs = rxTimeBs + config_->sampsPerSymbol * config_->symbolsPerFrame * 20;
+                    int r_tx = baseRadioSet_->radioTx(it, beaconbuff.data(), 2, txTimeBs);
```

Comment:
Similar comment about using a `flags` variable. In `macros.h`, we can define something like `kBurstEnd` and `kContinuous` constants and use them here.

Reply:
Done, use kStreamContinuous and kStreamEndBurst in include/macros.h

Though, eventually, we should probably use SOAPY_SDR_RX (flags=1) and SOAPY_SDR_END_BURST (flags=2)